### DISCUSSION TITLE
SSE with different LabelToNode instance

### DISCUSSION THREAD

**Discussion by cygri** [apache/jena#1271] [General]
This is a request for a way to use the `SSE` class with a `ParseHandler` that uses `LabelToNode.createUseLabelEncoded()` instead of the hardcoded `LabelToNode.createScopeGlobal()`. 
Perhaps:

- Add a new method `public static SSE.setParseHandler(ParseHandler handler)`.
- Add a method `public ParseHandlerPlain.setLabelToNode(LabelToNode bNodeLabels)` to set the `bNodeLabels` field (or make `emitBNode(...)` non-final so that it can be overridden to use a different `LabelToNode` instance).

Does this sound reasonable? I'm happy to submit a PR with these changes, or with any other API changes that would enable my goal.

**Comment by afs** [comment]
SSE is old. It ought to take a `FactoryRDF` (usually as part of a `ParserProfile`) to make nodes. These can have a specific label mapping.

What's the use case? SSE is primarily for Jena's tests.

**Reply by cygri** [reply]
Same use case, making objects for use in unit tests.

**Reply by afs** [reply]
`<_:ThisIsTheLabel>` is parsed to `_:ThisIsTheLabel`

It may print as `_:BThisIsTheLabel` in NT but the label is "ThisIsTheLabel".
